eos/saveddata/mode.py

```diff
@@ -53,3 +53,8 @@ def calculateModifiedAttributes(self, fit, runTime, forceProjected=False):
             for effect in self.item.effects.itervalues():
                 if effect.runTime == runTime and effect.activeByDefault:
                     effect.handler(fit, self, context=("module",))
+
+    def getValidCharges(self):
+        # Modes don't have charges, but it is queried for so return nothing.
+        validCharges = set()
+        return validCharges
```

Comment:
It still doesn't feel right to me. If modes ever get charges, then we simply add it (at that point we would probably just inherit from Module). Having it defined here for no purpose other than to work around a bug makes me thing we should conditionally account for it where it's called.

Also, going the route I suggested in another comment about giving the mode it's own context menu... context (*cringe*) would solve this, as we can then selectively determine if we even run the offending code (the module ammo picker). The current problem is that the context menu sees the mode as a module, and all modules are assumed to have charge properties. And this context menu runs because it things the item is a module, and thus assumes it has all the module stuff. Having a separate context for modes would be the best way to go IMO, and it's not difficult at all to do. :)

Let me know if you need additional help about it :D